operator/watchers/cilium_endpoint.go

```diff
@@ -154,3 +155,16 @@ func HasCE(ns, name string) (*cilium_api_v2.CiliumEndpoint, bool, error) {
 	cep := item.(*cilium_api_v2.CiliumEndpoint)
 	return cep, exists, nil
 }
+
+// HasCEWithIdentity returns true or false if the Cilium Endpoint store has
+// the given identity.
+// Once the CiliumEndpoint is migrated to use resources
+// the cepStore can be removed and used the local store.
+func HasCEWithIdentity(cepStore resource.Store[*cilium_api_v2.CiliumEndpoint], identity string) bool {
```

Comment:
I don't think this helper should live here in the CiliumEndpoint watcher file. 
This helper is using resource.Store[T], not the `CiliumEndpointStore` defined here. Also, the key for the indexer is the same string as the one for `CiliumEndpointStore`, but it should refer the `k8s.CiliumEndpointIndexIdentity` constant defined in `operator/k8s/resources.go` instead, as the cepStore argument is derived from that Resource.

Since this PR is not actually using `HasCEWithIdentity`, I suggest to remove the commit and introduce it later, where it will be clear where we need this and thus where this should live.

Reply:
Removed this commit from here.

A draft full implementation can be found https://github.com/cilium/cilium/pull/30356/. But please keep in mind that the tests might look different but the main idea should be illustrated fine.

Reply:
Thanks, that's useful. I suggest to link the draft implementation in the subsequent PRs too. :100: